### PR TITLE
Restructures the CollectionNavigation JavaScript functionality for paginating more than 10 online content components

### LINKED ISSUE #457: Collection-level online content tab displaying only 10 components
The "Online content" tab on a collection detail page displays components with digital objects when there are no collection-level digital objects. Currently, it is displaying only 10 components. 

**Acceptance criteria (for discussion/negotiation)**

* When on the Online content tab on a collection detail page and there are no collection-level display objects, display all of the components with digital objects.
* Consider providing some pagination on this display.

**Screenshot**

screen shot 2017-06-14 at 2 43 16 pm

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Resolves #457

### CODE REVIEW

File: app/assets/javascripts/arclight/collection_navigation.js [hunk 1]
```diff
@@ -44,47 +44,125 @@
...
+        }
+
+        // Ensure that the pagination elements are added to the AJAX-loaded
+        // content
+        if (paginationElements.length > 0) {
+          newDocs.append(paginationElements);
+
+          // Override the handlers to refresh the content
+          var paginationLinks = paginationElements.find('a')
+          paginationLinks.on('click', function(e) {
```
Comment (jrgriffiniii) [inline app/assets/javascripts/arclight/collection_navigation.js]: I don't really like the approach which I've proposed here, but I didn't want to proceed with further refactoring attempts.  The restructuring here so far was undertaken in order to try and take a recursive approach for invoking `$.ajax` for the AJAX requests in jQuery.

**Comment by jrgriffiniii** [comment]
https://travis-ci.org/projectblacklight/arclight/jobs/573922832 is failing due to XML validation errors.

**Comment by jrgriffiniii** [comment]
A number of acceptance tests were failing with the introduction of https://github.com/projectblacklight/arclight/pull/579/commits/9611a1f8f987fb82762db466890a669511365276, https://github.com/projectblacklight/arclight/pull/579/commits/12185ab19e3100b38b971850d414617b53b4c734 might resolve these and get this PR prepared for review.

**Comment by mejackreed** [comment]
Is it possible to add some fake online content components to existing fixtures to preempt some of the failures?

File: app/assets/javascripts/arclight/collection_navigation.js (hunk 1 above)
Comment (mejackreed) [reply app/assets/javascripts/arclight/collection_navigation.js]: Do we want to see if the upcoming content navigation has any implication on this? Or if we implemented ES6 support that would make life easier?

Comment (jrgriffiniii) [reply app/assets/javascripts/arclight/collection_navigation.js]: I defer to you and others on this.  `.render` calls and ES5 Classes such as `Paginator` were proposed with the hope of just preparing to broach this subject of ES6 Classes and Module exports.

Comment (jrgriffiniii) [reply app/assets/javascripts/arclight/collection_navigation.js]: I've found the following issues which might have design implications (potentially blocking this, or requiring significant changes):
- https://github.com/projectblacklight/arclight/issues/536
- https://github.com/projectblacklight/arclight/issues/514
- https://github.com/projectblacklight/arclight/issues/540
- https://github.com/projectblacklight/arclight/issues/534
- https://github.com/projectblacklight/arclight/issues/546

If there are others which might block this (or require changes), please let me know and I can address these.  Also, if it is preferred that I make an attempt at https://github.com/projectblacklight/arclight/issues/587, I definitely understand and can make this a priority.

Comment (ggeisler) [reply app/assets/javascripts/arclight/collection_navigation.js]: FWIW, I believe we're very close to finalizing our broad design change recommendations (I'll bring up in today's planning meeting). If we get agreement on those, we'll be able to start posting specific design updates to most or all of the tickets you noted above @jrgriffiniii.

Comment (jrgriffiniii) [reply app/assets/javascripts/arclight/collection_navigation.js]: Thank you @ggeisler, I will be certain to adjust this in response to those recommendations.  Also, please let me know if I can assist with implementing any of those.

Comment (mejackreed) [reply app/assets/javascripts/arclight/collection_navigation.js]: 🆒  so sounds like we have a good working way forward here, and we can just hold for a bit until we see the updated designs?

**Comment by jrgriffiniii** [comment]
ES6 integration with https://github.com/projectblacklight/arclight/pull/743 permits me to proceed with restructuring this as a proper PR.

**Comment by mejackreed** [comment]
@jrgriffiniii When looking into another issue, I noticed this: https://github.com/projectblacklight/arclight/blob/master/app/models/concerns/arclight/search_behavior.rb#L21

That could be extended to the `online_contents` view. While this may not solve the pagination problem, perhaps that is ok?

**Comment by jrgriffiniii** [comment]
> That could be extended to the online_contents view. While this may not solve the pagination problem, perhaps that is ok?

Sure, I can investigate this.

**Comment by jrgriffiniii** [comment]
> That could be extended to the online_contents view. While this may not solve the pagination problem, perhaps that is ok?

When I attempt this with https://github.com/projectblacklight/arclight/pull/579/commits/4b9bda334b0b7d2942907209c1f72f98b0b919fd I still see only 10 results retrieved, I will attempt to work with the `rows` parameter to try and further this approach.

**Comment by jrgriffiniii** [comment]
> When I attempt this with 4b9bda3 I still see only 10 results retrieved, I will attempt to work with the rows parameter to try and further this approach.

This was wrong, https://github.com/projectblacklight/arclight/pull/579/commits/dc1ef082a8a37690fd5f02669e8a6eb2ad862da5 indeed gets this working (as suggested by @mejackreed).  I will retain the original JavaScript and see about structuring some tests for this case.

**Pull request merged**